backend/apps/owasp/management/commands/owasp_aggregate_contributions.py

```diff
@@ -0,0 +1,418 @@
+            help="Number of days to look back for contributions (default: 365)",
+        )
+        parser.add_argument(
+            "--key",
+            type=str,
+            help="Specific chapter or project key to aggregate",
+        )
+        parser.add_argument(
+            "--offset",
+            type=int,
+            default=0,
+            help="Skip the first N entities",
+        )
+
+    def _aggregate_contribution_dates(
+        self,
+        queryset,
+        date_field: str,
+        contribution_map: dict[str, int],
+    ) -> None:
+        """Aggregate contribution dates from a queryset into the contribution map.
+
+        Args:
+            queryset: Django queryset to aggregate
+            date_field: Name of the date field to aggregate on
+            contribution_map: Dictionary to update with counts
+
+        """
+        dates = queryset.values_list(date_field, flat=True)
+        for date_value in dates:
+            if date_value:
+                date_key = date_value.date().isoformat()
+                contribution_map[date_key] = contribution_map.get(date_key, 0) + 1
+
+    def aggregate_chapter_contributions(
+        self,
+        chapter: Chapter,
+        start_date: datetime,
+    ) -> dict[str, int]:
+        """Aggregate contributions for a chapter.
+
+        Args:
+            chapter: Chapter instance
+            start_date: Start date for aggregation
+
+        Returns:
+            Dictionary mapping YYYY-MM-DD to contribution count
+
+        """
+        contribution_map: dict[str, int] = {}
+
+        if not chapter.owasp_repository:
+            return contribution_map
+
+        repository = chapter.owasp_repository
+
+        # Aggregate commits
+        self._aggregate_contribution_dates(
+            Commit.objects.filter(
+                repository=repository,
+                created_at__gte=start_date,
+            ),
+            "created_at",
+            contribution_map,
+        )
+
+        # Aggregate issues
+        self._aggregate_contribution_dates(
+            Issue.objects.filter(
+                repository=repository,
+                created_at__gte=start_date,
+            ),
+            "created_at",
+            contribution_map,
+        )
+
+        # Aggregate pull requests
+        self._aggregate_contribution_dates(
+            PullRequest.objects.filter(
+                repository=repository,
+                created_at__gte=start_date,
+            ),
+            "created_at",
+            contribution_map,
+        )
+
+        # Aggregate releases (exclude drafts)
+        self._aggregate_contribution_dates(
+            Release.objects.filter(
+                repository=repository,
+                published_at__gte=start_date,
+                is_draft=False,
+            ),
+            "published_at",
+            contribution_map,
+        )
+
+        return contribution_map
+
+    def aggregate_project_contributions(
```

Comment:
Most of the logic for chapters/projects is the same. Please reduce the code redundancy by refactoring it to universal methods for both chapters/projects